corehq/apps/locations/tests/test_locations.py

```diff
@@ -210,6 +251,130 @@ def compare(list1, list2):
+        self.assertEqual(
+            'teststate-Cases',
+            self.test_state.sql_location.case_sharing_group_object().name
+        )
+
+        # locations combined by forward slashes otherwise
+        self.assertEqual(
+            'teststate/testvillage/testoutlet-Cases',
+            self.test_outlet.sql_location.case_sharing_group_object().name
+        )
+
+        # reporting group is similar but has no ending
+        self.assertEqual(
+            'teststate/testvillage/testoutlet',
+            self.test_outlet.sql_location.reporting_group_object().name
+        )
+
+    def test_id_assignment(self):
+        # each should have the same id, but with a different prefix
+        self.assertEqual(
+            'locationgroup-' + self.test_outlet._id,
+            self.test_outlet.sql_location.case_sharing_group_object()._id
+        )
+        self.assertEqual(
+            'locationreportinggroup-' + self.test_outlet._id,
+            self.test_outlet.sql_location.reporting_group_object()._id
+        )
+
+    def test_group_properties(self):
+        # case sharing groups should ... be case sharing
+        self.assertTrue(
+            self.test_outlet.sql_location.case_sharing_group_object().case_sharing
+        )
+        self.assertFalse(
+            self.test_outlet.sql_location.case_sharing_group_object().reporting
+        )
+
+        # and reporting groups reporting
+        self.assertFalse(
+            self.test_outlet.sql_location.reporting_group_object().case_sharing
+        )
+        self.assertTrue(
+            self.test_outlet.sql_location.reporting_group_object().reporting
+        )
+
+        # both should set domain properly
+        self.assertEqual(
+            self.domain.name,
+            self.test_outlet.sql_location.reporting_group_object().domain
+        )
+        self.assertEqual(
+            self.domain.name,
+            self.test_outlet.sql_location.case_sharing_group_object().domain
+        )
+
+    def test_accessory_methods(self):
+        # we need to expose group id without building the group sometimes
+        # so lets make sure those match up
+        expected_id = self.loc.sql_location.case_sharing_group_object()._id
+        self.assertEqual(
+            expected_id,
+            self.loc.group_id
+        )
+
+    def test_not_real_groups(self):
+        # accessing a group object should not cause it to save
+        # in the DB
+        group_obj = self.test_outlet.sql_location.case_sharing_group_object()
+        with self.assertRaises(ResourceNotFound):
+            Group.get(group_obj._id)
+
+    def test_custom_data(self):
+        # need to put the location data on the
+        # group with a special prefix
+        self.loc.metadata = {
+            'foo': 'bar',
+            'fruit': 'banana'
+        }
+        self.loc.save()
+
+        self.assertDictEqual(
+            {
+                'commcare_location_type': self.loc.location_type,
```

Comment:
do we want to remove this prefix? i don't think there's any other way to get key/values into here